### PR TITLE
Experiment with client interface and get tests working

### PR DESCRIPTION
**Pull request by jlongster**

### CODE REVIEW

**Comment by jlongster** [comment]
Not ready to be merged, but putting a prototype up for discussion.

**Comment by jlongster** [comment]
Yay, I wrote a test! https://github.com/jlongster/debugger.html/blob/client/js/actions/tests/sources.js

I gave up on a perfect mapping of the gecko-dev xpcshell environment. Normally you don't require the head file, but it's automatically included for you. When we do the merge, we'll have to go through and convert that require statement to something else. But that's not a big deal; we can probably even use a codegen tool to automatically go through and do it.

Everything except the `require("../../util/test-head");` should "just work" when we move to the real xpcshell harness.

This is include in the client connection work because the whole point of moving that to be local to the actions is so that it's easy to stub it out. Now we just need to create this mock connection object: https://github.com/jlongster/debugger.html/blob/client/js/util/test-head.js All of the actions will reference that mock object.

**Comment by jlongster** [comment]
I'd really like it if tests were a focus for the next day. I'll talk to you tomorrow about this, Jason.

File: js/client.js [hunk 1]
```diff
@@ -4,33 +4,41 @@ const { DebuggerClient } = require("devtools/shared/client/main");
...
 
 function connectToTab(tab, onNewSource) {
   let deferred = promise.defer();
-  const options = { client, form: tab, chrome: false };
+  const options = { currentClient, form: tab, chrome: false };
 
   TargetFactory.forRemoteTab(options).then(target => {
     target.activeTab.attachThread({}, (res, threadClient) => {
       threadClient.resume();
       window.gThreadClient = threadClient;
```
Comment (jasonLaster) [inline js/client.js]: can we delete this now?

File: js/components/Accordion.js [hunk 2]
```diff
@@ -39,7 +39,8 @@ const Accordion = React.createClass({
       { className: "accordion" },
       this.props.items.map((item, i) => {
         return div(
-          { className: opened[i] ? "opened" : "" },
+          { className: opened[i] ? "opened" : "",
+            key: i },
```
Comment (jasonLaster) [inline js/components/Accordion.js]: :+1:

**Comment by jasonLaster** [comment]
merged!

**Pull request closed without merging**